doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -1409,6 +1412,10 @@ table further down lists the classes that that are handled plainly.
     Parses an open :token:`term`. Like :n:`constr` above, this class
     accepts a list of notation scopes with the same effects.
 
+  :n:`preterm {? ( {+, @scope_key } ) }`
+    Parses a non-typechecked :token:`term`. Like :n:`constr` above, this class
+    accepts a list of notation scopes with the same effects.
```

Comment:
```suggestion
  :n:`preterm {? ( {+, @scope_key } ) }`
    Parses a :token:`term` without immediately typechecking it. Like :n:`constr` above, this class
    accepts a list of notation scopes with the same effects.
```

Would you add a sentence explaining when the term would eventually be typechecked and why/when `preterm` is useful?  Maybe add a hyperlink to `preterm` (which I suggest above)?

"non-typechecked" seems imprecise; eventually it will be typechecked

Reply:
It is typechecked only when explicitly passed to `Constr.pretype`.  it's useful when you want to add more information before typechecking; for example, when you write `exact foo`, `foo` should not be typechecked immediately; instead `foo : <the type of the goal>` should be typechecked.

Reply:
Currently I'm not sure preterm is useful as it has too little API.

>"non-typechecked" seems imprecise; eventually it will be typechecked

Not if we don't call the typechecking API on it. Non-typechecked is fine IMO

Reply:
@SkySkimmer you get preterms in notations.

Reply:
> Non-typechecked is fine IMO

The problem is that AFAIK the doc doesn't say anything about when typechecking is done or skipped entirely or why this is a good thing.  As a non-expert, it seems strange that some things are not typechecked.  It would be a great improvement if you could provide a few sentences that describe when typechecking is done and when not.  Or if you give me the information in a couple bullet points, I'll work on the wording.

Reply:
Currently it is typechecked by passing to Constr.pretype. In the future there may be other APIs that do it.

Reply:
> it seems strange that some things are not typechecked

Typechecking is costly and slow, and may fail for ill-typed or incomplete terms that can nonetheless hold useful data about how to build well-typed terms.  It's useful to have access to this.